packages/shared/lib/core/wallet/utils/outputs/getStorageDepositFromOutput.ts

```diff
@@ -1,13 +1,22 @@
 import { OutputTypes } from '@iota/types'
 import { OUTPUT_TYPE_TREASURY, UNLOCK_CONDITION_STORAGE_DEPOSIT_RETURN } from '../../constants'
 
-export function getStorageDepositFromOutput(output: OutputTypes): number {
+export function getStorageDepositFromOutput(output: OutputTypes): {
+    storageDeposit: number
+    giftedStorageDeposit: number
+} {
     if (output && output?.type !== OUTPUT_TYPE_TREASURY) {
         for (const unlockCondition of output.unlockConditions) {
             if (unlockCondition?.type === UNLOCK_CONDITION_STORAGE_DEPOSIT_RETURN) {
-                return Number(unlockCondition.amount)
+                return { storageDeposit: Number(unlockCondition.amount), giftedStorageDeposit: 0 }
             }
         }
+        if (output?.nativeTokens?.length > 0 && Number(output?.amount) > 0) {
+            return { storageDeposit: 0, giftedStorageDeposit: Number(output?.amount) }
+        } else {
+            return { storageDeposit: 0, giftedStorageDeposit: 0 }
+        }
+    } else {
+        return { storageDeposit: 0, giftedStorageDeposit: 0 }
```

Comment:
You sent an article a while back about the negatives of the implicit else. so been adding else explicitly since unless its a guard at the top of the function.

Reply:
![Screenshot 2022-08-03 at 17 13 08](https://user-images.githubusercontent.com/89913787/182657705-596fe7e1-e6bc-47bd-b830-7cb0545a3ffc.png)
I cant do that because the compiler doesn't recognise the type narrowing using the find function, so I would need to add the extra if statement in anyway.

Reply:
I could cast it?

Reply:
Sorry for the ambiguous request. I thought about it and in this scenario it's a trade-off between code duplication (returning `{ storageDepost:0 ... }` twice)  and having explicit elses. I favor incurring a minor cost of readability over code duplication, but I'll leave it up to you.

Casting it seems like a good solution!

Reply:
Already made the change by the way